dotCMS/src/main/java/com/dotcms/rendering/js/JsFileSystem.java

```diff
@@ -0,0 +1,277 @@
+package com.dotcms.rendering.js;
+
+import com.dotmarketing.beans.Host;
+import com.dotmarketing.beans.Identifier;
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DoesNotExistException;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.contentlet.business.DotContentletStateException;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.fileassets.business.FileAsset;
+import com.dotmarketing.util.HostUtil;
+import com.dotmarketing.util.UtilMethods;
+import io.vavr.Tuple2;
+import org.apache.commons.compress.utils.SeekableInMemoryByteChannel;
+import org.graalvm.polyglot.io.FileSystem;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.URI;
+import java.nio.channels.SeekableByteChannel;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.AccessMode;
+import java.nio.file.DirectoryStream;
+import java.nio.file.LinkOption;
+import java.nio.file.NoSuchFileException;
+import java.nio.file.OpenOption;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.nio.file.attribute.FileAttribute;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Set;
+
+/**
+ * This File system implementation allows to the js context to access file stored into the
+ * it supports things such as:
+ * /application/js/modules/mymodules.js
+ * //demo.dotcms.com/application/js/modules/mymodules.js
+ * /javascript/modules/systemmodule.js
+ * dotCMS file system.
+ * @author jsanca
+ */
+
+// todo:, node: this is working but i do not like the implementation, probably we should do the transformation on the parse (cross site o current site) to file asset path
+    // and remove the hacks to allow by pass as a path the sites ones
+public class JsFileSystem implements FileSystem {
+
+    private static final String APPLICATION_ROOT_PATH = "/application/";
+    private static final String JAVASCRIPT_ROOT_PATH = "/javascript/";
+    public static final String DOT_SITE_TOKEN = "/dot-site-token/";
+    public static final String PATH_IS_ONLY_ALLOWED_ON = "Path is only allowed on: (";
+
+    public JsFileSystem() {
+    }
+
+    private Path checkAllowedPath(final Path path) {
```

Comment:
Please insure that we do not allow for directory traversal, e.g. we do not allow access to 
`../../../../../../../../etc/passwd` or even `../WEB-INF/whatever`